types/angular/angular-tests.ts

```diff
@@ -367,8 +367,8 @@ namespace TestQ {
         let result: angular.IPromise<TResult>;
         result = $q.resolve<TResult>(tResult);
         result = $q.resolve<TResult>(promiseTResult);
-        result = $q.resolve<TResult | TOther>(Math.random() > 0.5 ? tResult : promiseTOther);
-        result = $q.resolve(Math.random() > 0.5 ? tResult : promiseTOther);
+        let result2: angular.IPromise<TResult | TOther> = $q.resolve<TResult | TOther>(Math.random() > 0.5 ? tResult : promiseTOther);
+        let result3: angular.IPromise<TResult | TOther | angular.IPromise<TOther>> = $q.resolve(Math.random() > 0.5 ? tResult : promiseTOther);
```

Comment:
After much discussion with @andy-ms and @DanielRosenwasser, I think the right thing is to require the type parameters here. The compiler can't infer the correct type. I'll deprecate the two-parameter overload, too, since inference will never work and the single-argument example on line 370 is always clearer.